File: WebViewApp/app/src/main/java/com/example/webviewapp/contract/EditRecordContract.java
package com.example.webviewapp.contract;

import com.example.webviewapp.data.Record;

public interface EditRecordContract {

    interface View {
        void setConfirmButtonVisibility(Boolean isVisible);
    }

    interface Presenter {
        void checkContent(String title, String details);

        Record getData(long primaryKey);

        void updateRecord(Record record);
    }
}

File: WebViewApp/app/src/main/java/com/example/webviewapp/contract/EditRecordContract.kt
package com.example.webviewapp.contract

import com.example.webviewapp.data.Record

interface EditRecordContract {
    interface View {
        fun setConfirmButtonVisibility(isVisible: Boolean?)
    }

    interface Presenter {
        fun checkContent(title: String?, details: String?)
        fun getData(primaryKey: Long): Record?
        fun updateRecord(record: Record?)
    }
}